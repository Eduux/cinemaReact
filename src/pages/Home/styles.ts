import styled from 'styled-components';

import { container } from 'styles/mixins';

export const Container = styled.div`
  ${container};
`;
