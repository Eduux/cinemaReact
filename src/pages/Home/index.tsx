import React from 'react';

import { useMovies } from 'containers/movies';

import ListMovies from 'components/ListMovies';
import ErrorMessage from 'components/ErrorMessage';
import Loader from 'components/Loader';
import Pagination from 'components/Pagination';

import { Container } from './styles';

const Home: React.FC = () => {
  const {
    data: { list, loading, error, pagination },
    actions: { changePage },
  } = useMovies();

  return error ? (
    <Container>
      <ErrorMessage>{error}</ErrorMessage>
    </Container>
  ) : (
    <Container>
      {loading ? (
        <Loader />
      ) : (
        <>
          <ListMovies list={list} />
          <Pagination
            page={pagination.page}
            pages={pagination.pages}
            onChange={changePage}
          />
        </>
      )}
    </Container>
  );
};

export default Home;
