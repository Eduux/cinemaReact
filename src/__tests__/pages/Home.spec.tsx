import React from 'react';
import { render } from '@testing-library/react';

import Home from 'pages/Home';

import withMovies, { initialState } from 'containers/movies';

describe('Home page', () => {
  it('should render without data', () => {
    const Container = withMovies(Home, initialState);

    const result = render(<Container />);
    expect(result).toMatchSnapshot();
  });

  it('must render with error', () => {
    const Container = withMovies(Home, {
      ...initialState,
      error: 'Ocorreu um erro inesperado!',
    });

    const result = render(<Container />);
    expect(result).toMatchSnapshot();
  });

  it('must render with data', () => {
    const Container = withMovies(Home, {
      ...initialState,
      list: [
        {
          imdbID: '1',
          poster: 'a',
          title: 'a',
          type: 'a',
          year: '1960',
        },
      ],
    });

    const result = render(<Container />);
    expect(result).toMatchSnapshot();
  });

  it('must render with loading', () => {
    const Container = withMovies(Home, {
      ...initialState,
      loading: true,
    });

    const result = render(<Container />);
    expect(result).toMatchSnapshot();
  });
});
