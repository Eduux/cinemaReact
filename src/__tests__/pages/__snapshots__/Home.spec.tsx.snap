// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page must render with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-crHlIS laloAv"
      >
        <ul
          class="sc-iCfLBT cscthS"
        >
          <li
            class="sc-dkPtyc kIhNcZ"
          >
            <div
              class="sc-eCImvq dyuREx"
            >
              <img
                alt="a"
                class="sc-bdvvaa cLlfML"
                height="auto"
                loading="lazy"
                src="a"
                width="auto"
              />
            </div>
            <div
              class="sc-hKwCoD fRRmAz"
            >
              <strong
                class="sc-jRQAMF dxQwMq"
              >
                a
              </strong>
              <strong
                class="sc-jRQAMF dxQwMq"
              >
                a
              </strong>
              <strong
                class="sc-jRQAMF dxQwMq"
              >
                1960
              </strong>
              <div
                class="sc-gKckTs eMhRTy"
              >
                <button
                  class="sc-gsDJrp ClvND"
                  type="submit"
                >
                  Favoritar
                </button>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-crHlIS laloAv"
    >
      <ul
        class="sc-iCfLBT cscthS"
      >
        <li
          class="sc-dkPtyc kIhNcZ"
        >
          <div
            class="sc-eCImvq dyuREx"
          >
            <img
              alt="a"
              class="sc-bdvvaa cLlfML"
              height="auto"
              loading="lazy"
              src="a"
              width="auto"
            />
          </div>
          <div
            class="sc-hKwCoD fRRmAz"
          >
            <strong
              class="sc-jRQAMF dxQwMq"
            >
              a
            </strong>
            <strong
              class="sc-jRQAMF dxQwMq"
            >
              a
            </strong>
            <strong
              class="sc-jRQAMF dxQwMq"
            >
              1960
            </strong>
            <div
              class="sc-gKckTs eMhRTy"
            >
              <button
                class="sc-gsDJrp ClvND"
                type="submit"
              >
                Favoritar
              </button>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home page must render with error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-crHlIS laloAv"
      >
        <h2
          class="sc-pVTma iwwbOq"
        >
          Ocorreu um erro inesperado!
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-crHlIS laloAv"
    >
      <h2
        class="sc-pVTma iwwbOq"
      >
        Ocorreu um erro inesperado!
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home page must render with loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-crHlIS laloAv"
      >
        <div
          class="sc-jrQzUz kgXAAi"
        >
          <div
            class="sc-kDThTU fYywUt"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-crHlIS laloAv"
    >
      <div
        class="sc-jrQzUz kgXAAi"
      >
        <div
          class="sc-kDThTU fYywUt"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home page should render without data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-crHlIS laloAv"
      >
        <h2
          class="sc-furvIG dnKQvo"
        >
          Nenhum filme encontrado ou pesquisado.
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-crHlIS laloAv"
    >
      <h2
        class="sc-furvIG dnKQvo"
      >
        Nenhum filme encontrado ou pesquisado.
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
